docs/pages/access-controls/reference.mdx

```diff
@@ -418,14 +320,253 @@ spec:
+In a Teleport role resource, certain fields allow you to use functions and
+variables to configure a user's access. The functions and variables available
+for a field depend on the access controls that the field configures.
+
+### Access Request predicate functions
+
+The following variables and functions allow more fine-grained control over a
+user's permissions to submit and review Access Requests.
+
+#### Fields
+
+You can use Access Request predicate functions in the following fields. You can
+include these fields within either `allow` rules or `deny` rules:
+
+- `request.search_as_roles`
+- `request.roles`
+- `review_requests.preview_as_roles`
+- `review_requests.roles`
+- `review_requests.where`
+
+#### Functions
+
+Access Requests support the following role predicate functions:
+
+|Function|Description|
+|---|---|
+|`{{regexp.match(regexp)}}`|Returns `true` if the regular expression matches a user's role.|
+|`{{regexp.not_match(regexp)}}`|Returns `true` if the regular expression does not match a user's role.|
+
+Regular expressions support both wildcards (the `*` character) and [Go-style
+regular expressions](https://pkg.go.dev/regexp). If an expression begins with
+the `^` character and ends with the `$` character, Teleport will evaluate it as
+a regular expression. Otherwise, it will evaluate it as a wildcard expression.
+Wildcards match a sequence of one or more characters.
+
+#### Variables
+
+Teleport does not perform variable expansion on the values of the fields above.
+
+### Label expressions
+
+<Admonition type="warning">
+Label expressions are available starting in Teleport version `13.1.1`.
+All components of your Teleport cluster must be upgraded to version `13.1.1`
+or newer before you will be able to use label expressions.
+This includes the Auth Service and **all** Teleport agents.
+</Admonition>
+
+Teleport roles support matching resource labels with predicate expressions when
+you need to:
+
+- combine logic with OR and AND operators
+- perform matching on label keys
+- implement negative matches
+
+The following example role would allow access to any nodes labeled `env=staging`
+or labeled `team=<team>`, where `<team>` is one of the values of the user's
+`teams` trait:
+
+```yaml
+kind: role
+version: v7
+metadata:
+  name: example-role
+spec:
+  allow:
+    node_labels_expression: |
+      labels["env"] == "staging" ||
+      contains(user.spec.traits["teams"], labels["team"])
+```
+
+Typically you will only want to use one of `<kind>_labels` or
+`<kind>_labels_expression` in a single role, but you are allowed to specify
+both.
+If you have both in a deny rule, the matching is greedy, if either one matches
+access will be denied.
+In an allow rule, the matching is not greedy, if both are set they both have to
+match for access to be allowed.
+
+#### Fields
+
+The `<kind>_labels_expression` fields have the same purpose as the matching
+`<kind>_labels` fields, but support predicate expressions instead of label
```

Comment:
```suggestion
`<kind>_labels` fields, but support label expressions instead of label
```

I still think we need to call these by their proper name. Label expressions are different than the generic predicate language used to search and filter resources.